include/aikido/constraint/dart/IKRankingStrategy.hpp

```diff
@@ -0,0 +1,69 @@
+#ifndef AIKIDO_CONSTRAINT_DART_IKRANKINGSTRATEGY_HPP_
+#define AIKIDO_CONSTRAINT_DART_IKRANKINGSTRATEGY_HPP_
+
+#include <dart/dynamics/dynamics.hpp>
+#include "aikido/statespace/dart/MetaSkeletonStateSpace.hpp"
+
+namespace aikido {
+namespace constraint {
+namespace dart {
+
+class IKRankingStrategy
+{
+public:
+  // TODO (avk): Move this to protected?
+  // Read this in detail:
+  // 1. https://stackoverflow.com/questions/2290733/
+  // 1. initialize-parents-protected-members-with-initialization-list-c
+  // 2. https://stackoverflow.com/questions/18479295/
+  // 2. member-initializer-does-not-name-a-non-static-data-member-or-base-class
+  /// Constructor
+  ///
+  /// \param[in] metaSkeletonStateSpace Statespace of the skeleton.
+  /// \param[in] metaskeleton Metaskeleton of the robot.
+  /// \param[in] numIKSolution Number of IK solutions to rank.
+  IKRankingStrategy(
+      statespace::dart::ConstMetaSkeletonStateSpacePtr metaSkeletonStateSpace,
+      ::dart::dynamics::ConstMetaSkeletonPtr metaSkeleton,
+      std::size_t numIKSolutions);
+
+  /// Destructor
+  virtual ~IKRankingStrategy() = default;
+
+  /// Returns the statespace.
+  statespace::ConstStateSpacePtr getStateSpace() const;
+
+  // TODO(avk): What happens when the number of IK solutions
+  // is less than the number specified in constructor?
+
+  /// Returns the vector of ranked IK solutions.
+  std::vector<std::pair<statespace::StateSpace::State*, double>>&
+  getRankedIKSolutions();
+
+  /// Add IK solution to Ranker
+  /// Evaluates, creates a pair and stores it in the vector.
+  void addIKSolution(statespace::StateSpace::State* solution);
```

Comment:
I think it might be easier to require that we're given a vector of states, rather than adding one state at a time.

Reply:
Moreover, since there is no notion of removing IKs from the container, what you suggest is much neater than having this function, true! Good point!